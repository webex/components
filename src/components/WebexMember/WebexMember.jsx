import React from 'react';
import PropTypes from 'prop-types';

import {usePerson, useMembers} from '../hooks';
import WebexAvatar from '../WebexAvatar/WebexAvatar';
import {WEBEX_COMPONENTS_CLASS_PREFIX} from '../../constants';

/**
 * Displays a webex meeting member.
 *
 * @param {string} props.destinationType Type of destination of the membership roster
 * @param {string} props.destinationID  ID of the destination for which to get members
 * @param {string} props.personID  ID of the person for which to display avatar
 * @param {boolean} props.displayStatus  Whether or not to display the user's status
 *
 * @returns {object} JSX of the component
 */
export default function WebexMember({
  destinationType,
  destinationID,
  personID,
  displayStatus,
}) {
  const {firstName, lastName} = usePerson(personID);
  const members = useMembers(destinationID, destinationType);
  const member = members
    .find((itemMember) => itemMember.personID === personID);

  const isMuted = member && member.muted;

  return (
    <div className={`${WEBEX_COMPONENTS_CLASS_PREFIX}-member`}>
      <WebexAvatar personID={personID} displayStatus={displayStatus} />
      <div className="member-name">{`${firstName} ${lastName}`}</div>
      {isMuted && <div className="member-muted"><span className="icon icon-microphone-muted_24" /></div>}
    </div>
  );
}

WebexMember.propTypes = {
  destinationType: PropTypes.string.isRequired,
  destinationID: PropTypes.string.isRequired,
  personID: PropTypes.string.isRequired,
  displayStatus: PropTypes.bool,
};

WebexMember.defaultProps = {
  displayStatus: false,
};
