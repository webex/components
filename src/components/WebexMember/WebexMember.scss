@import '../../styles/variables';

.#{$WEBEX_COMPONENTS_CLASS_PREFIX}-member {
  display: flex;
  padding: 0.625rem 0 0.625rem 0;

  .member-name {
    @include body-fonts;
    align-items: center;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    padding-left: 0.25rem;
  }
}
