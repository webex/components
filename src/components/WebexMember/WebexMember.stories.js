import React from 'react';
import {storiesOf} from '@storybook/react';

import jsonData from '../../data';
import {WebexJSONAdapter} from '../../adapters';

import WebexMember from './WebexMember';

import {WebexDataProvider} from '..';

// Setup for the stories
const stories = storiesOf('Webex Member', module);
const webexAdapter = new WebexJSONAdapter(jsonData);

// Stories
stories.add('default', () => (
  <WebexDataProvider adapter={webexAdapter}>
    <WebexMember personID="default" />
  </WebexDataProvider>
));
