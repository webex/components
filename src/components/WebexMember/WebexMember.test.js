import React from 'react';

import WebexMember from './WebexMember';

jest.mock('../hooks/usePerson');

describe('Webex Member component', () => {
  describe('snapshot', () => {
    test('matches snapshot of webex member', () => {
      expect(shallow(<WebexMember personID="default" />)).toMatchSnapshot();
    });
  });
});
