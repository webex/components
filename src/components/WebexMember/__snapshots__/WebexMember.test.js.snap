// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webex Member component snapshot matches snapshot of webex member 1`] = `
<ContextTransform(ListItem)
  className="wxc-member"
>
  <WebexAvatar
    displayStatus={false}
    personID="default"
  />
  <div
    className="member-name"
  >
    Webex Component User
  </div>
</ContextTransform(ListItem)>
`;
